hoomd/Variant.h

```diff
@@ -50,27 +50,330 @@ class PYBIND11_EXPORT VariantConstant : public Variant
             {
             }
 
+        /// Return the value.
         Scalar operator()(uint64_t timestep)
             {
             return m_value;
             }
 
-        /// Set the value
+        /// Set the value.
         void setValue(Scalar value)
             {
             m_value = value;
             }
 
-        /// Get the value
+        /// Get the value.
         Scalar getValue()
             {
             return m_value;
             }
 
     protected:
-        /// The value
+        /// The value.
         Scalar m_value;
     };
 
+/** Ramp variant.
+
+    Variant that ramps linearly from A to B over a given number of steps.
+*/
+class PYBIND11_EXPORT VariantRamp : public Variant
+    {
+    public:
+
+        /** Construct a VariantRamp.
+
+            @param A The starting value.
+            @param B The ending value.
+            @param t_start The starting time step.
+            @param t_ramp The length of the ramp.
+        */
+        VariantRamp(Scalar A, Scalar B, uint64_t t_start, uint64_t t_ramp)
+            {
+            setA(A);
+            setB(B);
+            setTStart(t_start);
+            setTRamp(t_ramp);
+            }
+
+        /// Evaluate the ramp.
+        Scalar operator()(uint64_t timestep)
+            {
+            if (timestep < m_t_start)
+                {
+                return m_A;
+                }
+            else if (timestep < m_t_start + m_t_ramp)
+                {
+                // interpolate between A and B
+                uint64_t v = timestep - m_t_start;
+                double s = double(v) / double(m_t_ramp);
+                return m_B * s + m_A * (1.0 - s);
```

Comment:
This may not be important, but is there a reason we don't use this formulation
````C++
return m_A + (m_B - m_A) * s;
````
This has one less multiplication and it is just as simple to understand. We could even remove the subtraction by internally storing `m_B - m_A` as `delta` or something. I know this is not performance critical code just some quick thoughts I had that wouldn't reduce code maintainability or readability.

Who knows, the compiler may already be (probably is) smarter than me here.

Reply:
The compiler is actually not allowed to make these kinds of rearrangements (by default). The order of operations matters in floating point, so `m_B * s + m_A * (1.0 - s)` is not always equal to `m_A + (m_B - m_A) * s`. Specifically, when `s=1`, the first expression will give `m_B + 0 == m_B` while the second expression will compute `((m_A + m_B) - m_A)` which will not necessarily equal `m_B` due to round-off error.

According to Wikipedia (https://en.wikipedia.org/wiki/Linear_interpolation) the form you suggest is fine on platforms that implement FMA instructions. I don't think we can guarantee that on all platforms we support, so I'll keep the code as is. The extra one math operation per time step is not going to change performance.